src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -95,12 +115,20 @@ private static int getScaleSlot(int scale)
         throw new TpcdsException("scale was greater than max scale");
     }
 
-    private long computeCountUsingLinearScale(int scale)
+    private long computeCountUsingLinearScale(float scale)
     {
         long rowCount = 0;
-        int targetGB = scale;
+        float targetGB = scale;
+
+        if (scale < 1) {
+            rowCount = Math.round(scale * getRowCountForScale(DEFINED_SCALES[1]));
+            if (rowCount == 0) {
+                return 1;
+            }
+            return rowCount;
+        }
 
-        for (int i = DEFINED_SCALES.length - 1; i >= 0; i--) {  // work from large scales down
+        for (int i = DEFINED_SCALES.length - 1; i > 0; i--) {  // work from large scales down
```

Comment:
to make sure I understand: for decimals > 1, it doesn't do anything different from if it had been an integer, but that's because it's not very precise with the way it does linear scaling to begin with.  Is that right?